src/averageAges.js

```diff
@@ -37,7 +41,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(woman => (
+    withChildren
+      ? people.find(child => woman.name === child.mother)
+      : woman.sex === 'f'
+  ));
+  const womenAges = women.map(woman => woman.died - woman.born);
+  const averageWomenAge = womenAges
+    .reduce((sum, age) => sum + age, 0) / women.length;
```

Comment:
again. redundant map iteration. 
get age in reduse

Reply:
fixed!